src/Stryker.Core/Stryker.Core/Initialisation/InputFileResolver.cs

```diff
@@ -143,11 +146,25 @@ private void MarkInputFilesAsExcluded(FolderComposite root, List<string> pathsTo
                 }
             }
         }
+        
+        public string FindSpecificTestProjectFile(string currentDirectory, string testProjectNameFilter)
+        {
+            var testProjectFile = _fileSystem.Directory.GetFiles(currentDirectory, $"{testProjectNameFilter}");
+
+            if (!testProjectFile.Any())
+            {
+                throw new StrykerInputException(
+                    $"No .csproj file found, matching testProjectParameter: {testProjectNameFilter}");
+            }
+            
+            return testProjectFile.First();
```

Comment:
If we find more than one project matching the name filter we should throw an input exception. We should not make assumptions about the first we find being the correct one. This will lead to unexpected results for users..

Reply:
Good catch